reporter/v2/pkg/reporter/reconcile_task.go

```diff
@@ -101,6 +101,67 @@ func (r *ReconcileTask) run(ctx context.Context) error {
 			logger.Error(err, "error uploading files from data service", details...)
 			return err
 		}
+	} else if !r.CanRunUploadTask(ctx) {
```

Comment:
@zach-source is this the right approach to update the report statuses ?

Reply:
```suggestion
	} else if r.CanRunUploadTask(ctx) {
```